### PR TITLE
Fix prefetching lowering issue.

### PR DESCRIPTION
**Pull request by chengjunlu**
Fix two prefetching op lowering issues on fallback path:

- The element bits size is wrong. We need to use the element bits size as same as the tensor scalar type.
- The wrong DIM id is used in rounding the offset to fit into the tensor shape if size is too small.

### CODE REVIEW

**Comment by whitneywhtsang** [comment]
Can we still generate with OpenCL builtins?

**Comment by chengjunlu** [comment]
> Can we still generate with OpenCL builtins?

Some required variation of the OCL prefetching is not there. I haven't the full lists yet. Let me collect the information.

**Comment by whitneywhtsang** [comment]
> > Can we still generate with OpenCL builtins?
> 
> Some required variation of the OCL prefetching is not there. I haven't the full lists yet. Let me collect the information.

Is it possible to use the existing OCL prefetching variations, by changing our lowering algorithm?

**Comment by chengjunlu** [comment]
> > > Can we still generate with OpenCL builtins?
> > 
> > 
> > Some required variation of the OCL prefetching is not there. I haven't the full lists yet. Let me collect the information.
> 
> Is it possible to use the existing OCL prefetching variations, by changing our lowering algorithm?

It is possible. But need to double confirm whether it impacts the performance.

**Comment by whitneywhtsang** [comment]
> It is possible. But need to double confirm whether it impacts the performance.

Sounds good, let's double confirm that first.

**Comment by chengjunlu** [comment]
> > It is possible. But need to double confirm whether it impacts the performance.
> 
> Sounds good, let's double confirm that first.

Changes have been updated. No performance regression found.

File: third_party/intel/lib/Dialect/TritonGEN/IR/TritonGENOps.cpp
Comment (whitneywhtsang) [inline third_party/intel/lib/Dialect/TritonGEN/IR/TritonGENOps.cpp:1]: should we now remove the changes to verifier?

Comment (chengjunlu) [reply third_party/intel/lib/Dialect/TritonGEN/IR/TritonGENOps.cpp:1]: I remove the changes in the TritonGen to make the minimal changes.
Let's change the TritonGen verifier if a real case that requires a new prefetching shape.

File: third_party/intel/lib/TritonIntelGPUToLLVM/LoadStoreOpToLLVM.cpp [hunk 1]
```diff
@@ -247,18 +233,47 @@ struct PrefetchOpConversion
...
+        // element.
+        vBlocks = 2;
+        tileWidthInElem = 16;
+      } else
+        vBlocks = 1;
+      break;
+    default:
+      // for the default case.
+      vBlocks = 1;
+    }
```
Comment (whitneywhtsang) [inline third_party/intel/lib/TritonIntelGPUToLLVM/LoadStoreOpToLLVM.cpp:276]:
```suggestion
    unsigned vBlocks = 1;
    switch (elemSizeInBits) {
    case 8:
      if (tileWidthInElem == 64) {
        // OCL interface supports 8b_?r32x2c for 64 bytes per row of 8 bits
        // element.
        vBlocks = 2;
        tileWidthInElem = 32;
      }
      break;
    case 16:
      if (tileWidthInElem == 32) {
        // OCL interface supports 16b_?r16x2c for 64 bytes per row of 16 bits
        // element.
        vBlocks = 2;
        tileWidthInElem = 16;
      }
      break;
    }
```

**Review by whitneywhtsang — APPROVED** [review]

**Pull request merged**